homeassistant/components/sensor/uber.py

```diff
@@ -0,0 +1,247 @@
+"""
+Support for the Uber API.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.uber/
+"""
+import logging
+from datetime import timedelta
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+REQUIREMENTS = ['https://github.com/denismakogon/rides-python-sdk/archive/'
+                'py3-support.zip#'
+                'uber_rides==0.1.2-dev']
+
+ICON = 'mdi:taxi'
+
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=30)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Uber sensor."""
+    if None in (config.get('start_latitude'), config.get('start_longitude')):
+        _LOGGER.error(
+            "You must set start latitude and longitude to use the Uber sensor!"
+        )
+        return False
+
+    if config.get('server_token') is None:
+        _LOGGER.error("You must set a server_token to use the Uber sensor!")
+        return False
+
+    from uber_rides.session import Session
+
+    session = Session(server_token=config.get('server_token'))
+
+    wanted_product_ids = config.get('product_ids')
+
+    dev = []
+    start_lat_long = LatLong(config.get('start_latitude'),
+                             config.get('start_longitude'))
+    end_lat_long = LatLong(config.get('end_latitude'),
+                           config.get('end_longitude'))
+    timeandpriceest = UberEstimate(session, start_lat_long, end_lat_long)
+    for product_id, product in timeandpriceest.products.items():
+        if wanted_product_ids is not None and product_id in wanted_product_ids:
```

Comment:
Two things:

You probably want to change this to `wanted_product_ids is None or …` so you can drop the `elif`.

To avoid a lot of indentation, consider a guard clause:

```python
for …:
    if wanted_product_ids is not None and product_id not in wanted_product_ids:
        continue
    dev.append(…)
```